code/modules/projectiles/guns/projectile/sniper.dm

```diff
@@ -0,0 +1,175 @@
+
+/obj/item/weapon/gun/projectile/sniper_rifle
+	name = "sniper rifle"
+	desc = "the kind of gun that will leave you crying for mummy before you even realise your leg's missing"
+	icon_state = "moistnugget"
+	item_state = "moistnugget"
+	recoil = 2
+	heavy_weapon = 1
+	mag_type = /obj/item/ammo_box/magazine/sniper_rounds
+	fire_delay = 40
+	origin_tech = "combat=8"
+	can_unsuppress = 1
+	can_suppress = 1
+	w_class = 3
+	var/zoomed = FALSE
+	var/zoom_amt = 7
+	var/datum/action/sniper_zoom/azoom
+
+/obj/item/weapon/gun/projectile/sniper_rifle/syndicate
+	name = "syndicate sniper rifle"
+	desc = "syndicate flavoured sniper rifle, it packs quite a punch, a punch to your face"
+	pin = /obj/item/device/firing_pin/implant/pindicate
+	origin_tech = "combat=8;syndicate=4"
+
+/obj/item/weapon/gun/projectile/sniper_rifle/New()
+	..()
+	azoom = new()
+	azoom.rifle = src
+
+/obj/item/weapon/gun/projectile/sniper_rifle/dropped(mob/living/user)
+	zoom(user,FALSE)
+	azoom.Remove(user)
+
+/obj/item/weapon/gun/projectile/sniper_rifle/pickup(mob/living/user)
+	azoom.Grant(user)
+
+/obj/item/weapon/gun/projectile/sniper_rifle/proc/zoom(mob/living/user, forced_zoom)
```

Comment:
I would suggest this should probably be a generic projectile proc that can be added to *any* gun--energy, projectile, or otherwise---that way, if people want to have a secondary rifle that has a similar function, they can---likewise, if there's ever an R&D item like "Advanced Scope" or something, it could easily be added to any gun.